### PR TITLE
16466 - Fix outputs for corrections. Fix data missing for certificateOfNameCh…

### PR DESCRIPTION
**Pull request by seeker25**
…ange

*Issue #:* 

*Description of changes:*
https://preview.uxpin.com/53053d7aeb85fb29915f7afa596429aea4f2a964#/pages/163554292/simulate/sitemap

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: legal-api/src/legal_api/reports/report.py [hunk 1]
```diff
@@ -476,6 +474,9 @@ def _format_name_change_data(self, filing):
         if self._filing.filing_type == 'alteration':
             from_legal_name = meta_data.get('alteration', {}).get('fromLegalName')
             to_legal_name = meta_data.get('alteration', {}).get('toLegalName')
+        if self._filing.filing_type == 'correction':
+            from_legal_name = meta_data.get('correction', {}).get('fromLegalName')
```
Comment (seeker25) [inline legal-api/src/legal_api/reports/report.py:478]: Missing data for changeOfName for corrections:

image

File: legal-api/src/legal_api/core/filing_helper.py [hunk 2]
```diff
@@ -1,27 +1,14 @@
...
-    # Avoid circular import.
-    from legal_api.models import Filing  # pylint: disable=import-outside-toplevel
-    corrected_filing_type = filing['correction'].get('correctedFilingType')
-
-    if isinstance(business, Business) and business.legal_type != Business.LegalTypes.COOP.value:
-        return False
-    if isinstance(business, dict) and business.get('legalType') != Business.LegalTypes.COOP.value:
-        return False
-    if corrected_filing_type == 'specialResolution':
+    sr_correction_keys = ['rulesInResolution', 'resolution', 'rulesFileKey',
```
Comment (seeker25) [inline legal-api/src/legal_api/core/filing_helper.py:7]: Reworked this, so we don't care about parent filings

File: queue_services/entity-emailer/src/entity_emailer/email_processors/correction_notification.py [hunk 3]
```diff
@@ -230,12 +228,10 @@ def process(email_info: dict, token: str) -> Optional[dict]:  # pylint: disable=
...
             return None
     elif is_special_resolution_correction(
-            filing.filing_json['filing'],
-            business,
-            filing
+            filing.filing_json['filing']
     ):
         prefix = 'CP-SR'
-        name_changed = filing.filing_json['filing']['correction'].get('nameRequest', {})
+        name_changed = 'requestType' in filing.filing_json['filing'].get('correction', {}).get('nameRequest', {})
```
Comment (seeker25) [inline queue_services/entity-emailer/src/entity_emailer/email_processors/correction_notification.py:234]: Consistent with above, requestType CHG is included on changeOfName for a correction

File: queue_services/entity-emailer/src/entity_emailer/email_templates/CP-SR-CRCTN-COMPLETED.html [hunk 4]
```diff
@@ -19,7 +19,7 @@
           [[header.html]]
 
           <div class="container">
-            <p class="title-message bold">You have successfully filed a correction with the BC Business Registry.</p>
+            <p class="title-message bold">The correction filing for {{business.legalName}} is now effective.</p>
```
Comment (seeker25) [inline queue_services/entity-emailer/src/entity_emailer/email_templates/CP-SR-CRCTN-COMPLETED.html:22]: Feedback from Linda

File: queue_services/entity-emailer/src/entity_emailer/email_templates/CP-SR-CRCTN-PAID.html [hunk 5]
```diff
@@ -19,7 +19,6 @@
           [[header.html]]
 
           <div class="container">
-            <p class="title-message bold">You have successfully filed a correction with the BC Business Registry.</p>
```
Comment (seeker25) [inline queue_services/entity-emailer/src/entity_emailer/email_templates/CP-SR-CRCTN-PAID.html:22]: Feedback from Linda

File: legal-api/src/legal_api/reports/report.py [hunk 6]
```diff
@@ -950,12 +949,19 @@ def _format_special_resolution(self, filing):
...
+            to_legal_name = meta_data.get('changeOfName', {}).get('toLegalName')
+            if prev_legal_name and to_legal_name and prev_legal_name != to_legal_name:
+                filing['fromLegalName'] = prev_legal_name
+                filing['toLegalName'] = to_legal_name
+                filing['nrNumber'] = filing.get('changeOfName').get('nameRequest', {}).get('nrNumber', None)
+        elif filing_source == 'correction':
+            prev_legal_name = meta_data.get(filing_source, {}).get('fromLegalName')
+            to_legal_name = meta_data.get(filing_source, {}).get('toLegalName')
+            if prev_legal_name and to_legal_name and prev_legal_name != to_legal_name:
+                filing['fromLegalName'] = meta_data.get(filing_source, {}).get('fromLegalName')
```
Comment (seeker25) [inline legal-api/src/legal_api/reports/report.py:963]: image

File: queue_services/entity-filer/src/entity_filer/filing_processors/filing_components/rules_and_memorandum.py [hunk 7]
```diff
@@ -39,7 +39,7 @@ def update_rules(
         return None
 
     rules_file = MinioService.get_file(rules_file_key)
-    replace_file_with_certified_copy(rules_file.data, business, rules_file_key, business.founding_date, file_name)
+    replace_file_with_certified_copy(rules_file.data, business, rules_file_key, filing.filing_date, file_name)
```
Comment (seeker25) [inline queue_services/entity-filer/src/entity_filer/filing_processors/filing_components/rules_and_memorandum.py:42]: image

File: legal-api/src/legal_api/core/filing_helper.py [hunk 8]
```diff
@@ -1,27 +1,14 @@
...
+                          'memorandumInResolution', 'associationType']
+    for key in sr_correction_keys:
+        if key in filing.get('correction'):
+            return True
+    if 'requestType' in filing.get('correction', {}).get('nameRequest', {}):
```
Comment (seeker25) [inline legal-api/src/legal_api/core/filing_helper.py:12]: Let me know if this is a valid way of detecting a changeOfName in a correction

File: legal-api/src/legal_api/services/pdf_service.py [hunk 9]
```diff
@@ -21,9 +24,20 @@
 from reportlab.pdfbase.ttfonts import TTFont
 from reportlab.pdfgen import canvas
 
+from legal_api.reports.registrar_meta import RegistrarInfo
 from legal_api.utils.legislation_datetime import LegislationDatetime
 
 
+@dataclass
+class RegistrarStampData:
```
Comment (seeker25) [inline legal-api/src/legal_api/services/pdf_service.py:32]: Created so we can pass more than 4 arguments at once without making the function have way too many parameters

File: legal-api/src/legal_api/services/pdf_service.py [hunk 10]
```diff
@@ -56,26 +70,30 @@ def stamp_pdf(input_pdf, watermark, only_first_page=True):
...
                       mask='auto')
 
-        text = 'Filed on ' + LegislationDatetime.format_as_report_string(incorp_date)
-        if file_name:
-            text += '\nFile Name: ' + file_name
-        text += '\nIncorporation Number: ' + incorp_num
+        certify_date = LegislationDatetime.as_legislation_timezone(data.certify_date)
+        text = 'Filed on ' + LegislationDatetime.format_as_report_string(certify_date)
+        if data.is_correction:
+            text += ' (Corrected)'
```
Comment (seeker25) [inline legal-api/src/legal_api/services/pdf_service.py:93]: image

File: queue_services/entity-emailer/requirements.txt [hunk 11]
```diff
@@ -79,5 +79,5 @@ Werkzeug==1.0.1
 yarl==1.8.2
 zipp==3.15.0
 git+https://github.com/bcgov/business-schemas.git@2.18.10#egg=registry_schemas
-git+https://github.com/bcgov/lear.git#egg=legal_api&subdirectory=legal-api
+git+https://github.com/seeker25/lear.git@16466___#egg=legal_api&subdirectory=legal-api
```
Comment (seeker25) [inline queue_services/entity-emailer/requirements.txt:82]: Will revert

**Comment by seeker25** [comment]
image

Still to go -^ + linting + testing fixes

File: queue_services/entity-filer/requirements.txt [hunk 12]
```diff
@@ -24,5 +24,5 @@ minio==7.0.2
 PyPDF2==1.26.0
 reportlab==3.6.1
 git+https://github.com/bcgov/business-schemas.git@2.18.10#egg=registry_schemas
-git+https://github.com/bcgov/lear.git#egg=legal_api&subdirectory=legal-api
+git+https://github.com/seeker25/lear.git@16466___#egg=legal_api&subdirectory=legal-api
```
Comment (seeker25) [inline queue_services/entity-filer/requirements.txt:27]: Will revert

File: queue_services/entity-filer/src/entity_filer/filing_processors/filing_components/correction.py [hunk 13]
```diff
@@ -134,6 +136,15 @@ def correct_business_data(business: Business,  # pylint: disable=too-many-locals
             filing_meta.correction = {**filing_meta.correction,
                                       **{'uploadNewRules': True}}
 
+    with suppress(IndexError, KeyError, TypeError):
+        if dpath.util.get(correction_filing, '/correction/memorandumInResolution'):
+            filing_meta.correction = {**filing_meta.correction,
+                                      **{'memorandumInResolution': True}}
```
Comment (seeker25) [inline queue_services/entity-filer/src/entity_filer/filing_processors/filing_components/correction.py:142]: Need these in, so we don't rely on filing_data for anything outside the filer

**Comment by seeker25** [comment]
Reports should show up in Postman correctly now:

image

**Comment by seeker25** [comment]
I'd like to add some more unit tests for the documents, but I'm way over time on this one.

**Comment by seeker25** [comment]
I tried this on DEV, all seemed to work - except for the emails, working with Jia tomorrow to figure that out

File: legal-api/src/legal_api/reports/report.py [hunk 14]
```diff
@@ -58,7 +58,11 @@ def _get_static_report(self):
         document_type = ReportMeta.static_reports[self._report_key]['documentType']
         document: Document = self._filing.documents.filter(Document.type == document_type).first()
         response = MinioService.get_file(document.file_key)
-        return response.data, response.status
+        return current_app.response_class(
+            response=response.data,
+            status=response.status,
+            mimetype='application/pdf'
```
Comment (argush3) [inline legal-api/src/legal_api/reports/report.py:64]: thanks for adding the `mimetype`!

**Review by argush3 — APPROVED** [review]
nice work!

**Review by vysakh-menon-aot — APPROVED** [review]

**Pull request merged**